apps/backend/src/modules/points-of-interest/points-of-interest.service.ts

```diff
@@ -0,0 +1,112 @@
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { type CollectionResult, type Service } from "~/libs/types/types.js";
+import { PointsOfInterestEntity } from "~/modules/points-of-interest/points-of-interest.entity.js";
+import { type PointsOfInterestRepository } from "~/modules/points-of-interest/points-of-interest.repository.js";
+
+import { PointOfInterestExceptionMessage } from "./libs/enums/enums.js";
+import { PointOfInterestError } from "./libs/exceptions/exceptions.js";
+import {
+	type PointsOfInterestRequestDto,
+	type PointsOfInterestResponseDto,
+} from "./libs/types/type.js";
+
+class PointsOfInterestService implements Service {
+	private pointsOfInterestRepository: PointsOfInterestRepository;
+
+	public constructor(pointsOfInterestRepository: PointsOfInterestRepository) {
+		this.pointsOfInterestRepository = pointsOfInterestRepository;
+	}
+
+	public async create(
+		payload: PointsOfInterestRequestDto,
+	): Promise<PointsOfInterestResponseDto> {
+		await this.ensureNameIsUnique(payload.name);
+
+		const { name } = payload;
+
+		const item = await this.pointsOfInterestRepository.create(
+			PointsOfInterestEntity.initializeNew({
+				name,
+			}),
+		);
+
+		return item.toObject();
+	}
+
+	public async delete(id: number): Promise<boolean> {
+		const deleted = await this.pointsOfInterestRepository.delete(id);
```

Comment:
boolean properties should be named using is/has/should prefixes
```suggestion
		const isDeleted = await this.pointsOfInterestRepository.delete(id);
```